### PR TITLE
APID 2.15.0: Changes for the GitHub integration

### PR DESCRIPTION
**Pull request by rddheath**
Changed files:
[Create an API Specification with the Text Editor](https://github.com/mulesoft/docs-design-center/pull/139/files#diff-9781587b9b0f43c35826839cb4c1bdf3)
[Create and Publish a RAML API Fragment in the Text Editor in API Designer](https://github.com/mulesoft/docs-design-center/pull/139/files#diff-6cea404cf53fe669905557358b20fd33)

New files:
[Storing Projects in Design Center](https://github.com/mulesoft/docs-design-center/pull/139/files#diff-a2c9178cd18a39bc99d07ea8b8f7edfb)
[Storing Projects in GitHub](https://github.com/mulesoft/docs-design-center/pull/139/files#diff-84fff4f60f429cf31385cd9ee0c9c742)

### CODE REVIEW

**Review by carowright — DISMISSED** [review]

**Review by harshakirank — DISMISSED** [review]
LGTM

File: modules/ROOT/pages/design-storage-dc.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+include::_attributes.adoc[]
+endif::[]
+:page-aliases: design-branching.adoc
+
+:imagesdir: ../assets/images
+
+By default, the projects that you work on in API Designer are saved in Design Center. Only other users in your Anypoint Platform business organization can edit such projects.
+
+
+When you store in Design Center a project that you work on in API Designer's text editor (either an API-specification project or an API-fragment project), the project is saved to the branch `master` by default. You can create branches of the project if you do not want to change the content that is in the `master` branch. You can use all of the text editor's features when you are working in a branch. You can even publish the content of the branch as an asset in Anypoint Exchange.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-dc.adoc]: Is it accurate to say "You can create branches of the project if..." or should it say "You can create a branch of `master` if..."?

File: modules/ROOT/pages/design-storage-dc.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+[IMPORTANT]
+====
+You cannot merge changes made in one branch into another branch. For example, if you create the branch `prod-api-1` from your default branch, you cannot merge your changes in `prod-api-1` into the default branch. If you create the branch `prod-api-2` from `prod-api-1`, you cannot merge your changes in `prod-api-2` into `prod-api-1`. Branches are always independent of each other.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-dc.adoc]: Do we need to clarify "parent" branches or "origin" branches?
```suggestion
After you create one branch from another and modify it, you cannot merge those changes into the original branch. For example, if you create the branch `prod-api-1` from your default branch, you cannot merge your changes in `prod-api-1` into the default branch. If you create the branch `prod-api-2` from `prod-api-1`, you cannot merge your changes in `prod-api-2` into `prod-api-1`. Branches are always independent of each other.
```

File: modules/ROOT/pages/design-storage-dc.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+====
+
+When you create a branch, the text editor copies into the new branch the current content of the branch that you are in. For example, if you are in the `master` branch and you create the branch `branch-1`, the current content of `master` is copied into `branch-1`. If you subsequently create the branch `branch-2`, the current content of `branch-1` is copied into `branch-2`.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-dc.adoc]: Is this explanation necessary? Won't the audience understand what "creating a branch" entails? I do, and I am not as technical as they.

File: modules/ROOT/pages/design-storage-dc.adoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Even if there are branches in a project, when you open that project the editor opens it to the `master` branch. If you want to return to a branch that you were working in, you must switch to that branch.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-dc.adoc]:
```suggestion
Regardless of how many branches a project has, when you open that project, the editor opens it to the `master` branch. If you want to return to a branch that you were working in, you must switch to that branch.
```

File: modules/ROOT/pages/design-storage-dc.adoc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+To see which branch you are currently working in, look above the central pane of the text editor. Click the down arrow to switch to a branch or to the `master` branch. In the menu that appears, you can also create a branch.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-dc.adoc]:
```suggestion
To see which branch you are currently working in, look above the central pane of the text editor. Click the down arrow to expand the list of branches to switch to another branch, including the `master` branch. You can also create a branch from this list.
```

File: modules/ROOT/pages/design-storage-dc.adoc [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+.The controls for working with branches appear above the central pane.
+image:apid-dc-branching.png[The controls for working with branches.]
+
+<1> The names of the project and branch that you are working in.
+<2> The name of the branch that you are working in.
+<3> The list of branches, other than the branch that you are working in.
+<4> The icon to click to delete a branch.
+<3> The controls for creating a new branch.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-dc.adoc]: Did you intend the two entries for <3>?

File: modules/ROOT/pages/design-storage-github.adoc [hunk 7]
```diff
@@ -0,0 +1,99 @@
+= Storing Projects in GitHub
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+:imagesdir: ../assets/images
+
+By default, projects that you create in API Designer are saved in Design Center. However, you can save a project as a repository in GitHub.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]: Should we make clear in the previous text that although you have these two options, the default is Design Center?

File: modules/ROOT/pages/design-storage-github.adoc [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+image:apid-flow-for-GitHub.png[The steps for enabling and using the support for GitHub]
+
+As depicted on the far left of the diagram, one person with _Design Center Developer_ permission in your APBO installs the MuleSoft application in GitHub organization X. This person installs the application as part of creating the first API Designer project in the APBO to use GitHub.
+
+. Click *Create* to start creating a project in the text editor.
++
+You can create either an API-specification project that you want to work on in the text editor, or you can create an API-fragment project. The visual editor stores projects in Design Center.
+. Install the MuleSoft application in a GitHub organization.
++
+Click the *Install from GitHub* option to start the process for installing the MuleSoft application in a GitHub organization. As part of the process, you log into GitHub and select the organization to install the application in.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]: log in to:
```suggestion
Click the *Install from GitHub* option to start the process for installing the MuleSoft application in a GitHub organization. As part of the process, you log in to GitHub and select the organization to install the application in.
```

File: modules/ROOT/pages/design-storage-github.adoc [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
++
+After you finish the installation, you return to the dialog for creating a project.
+. In the *Save location* field, select your GitHub ID for the organization that you installed the MuleSoft application in.
+. Finish filling in the fields of the dialog and click the *Create* button.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]:
```suggestion
. Complete the fields of the dialog and click the *Create* button.
```

File: modules/ROOT/pages/design-storage-github.adoc [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+After the one-time installation, all developers in the APBO that have the _Design Center Developer_ permission and access to GitHub organization X can also create API Designer projects as repositories in that GitHub organization. They follow the same steps, but without following the steps of the installation procedure.
+
+Developers who do not have access to GitHub organization X, of course, cannot create projects as repositories in that organization, and they cannot edit projects that are saved there. However, developers who do have access to GitHub organization X but are not in the APBO can edit the repositories and create repositories to import into API Designer as projects, even though they cannot create projects through Design Center.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]:
```suggestion
Developers who do not have access to GitHub organization X cannot create projects as repositories in that organization, and they cannot edit projects that are saved there. However, developers who do have access to GitHub organization X but are not in the APBO can edit the repositories and create repositories to import into API Designer as projects, even though they cannot create projects through Design Center.
```

File: modules/ROOT/pages/design-storage-github.adoc [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+You can import an API-specification project or an API-fragment project that is stored as a repository in GitHub, whether you created the project in the text editor in API Designer or in an editor outside of API Designer.
+
+. Open Design Center from the homepage of Anypoint Platform.
+. On the *Projects* page in Design Center, click *Create* and then click *Create API Specification*.
+. Click *Import* near the top of the dialog.
++
+The dialog lists the projects that you can import from the GitHub organizations that your user ID belongs to.
+. Select the project that you want to import.
++
+If the project does not include an exchange.json file that was generated by the text editor in API Designer, a dialog is opened to ask you to specify which file is the root file of the project.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]:
```suggestion
If the project does not include an `exchange.json` file that was generated by the text editor in API Designer, a dialog is opened to ask you to specify which file is the root file of the project.
```

File: modules/ROOT/pages/design-storage-github.adoc [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+The project is imported and opened in the text editor.
+
+== Save an Existing API Designer Project as a Repository in GitHub
+
+If you have one or more API Designer projects that are saved in Design Center and you would like to save them instead in GitHub, follow these steps to migrate each one:
+
+. Export a copy of a project from Design Center.
+. Create in GitHub a repo for the copy of the project, but give the repo a name that is different from the name of the original project.
+. Extract the content of the downloaded ZIP file.
+. In the exchange.json file, edit the `name` field to match the name of the repo.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]:
```suggestion
. In the `exchange.json` file, edit the `name` field to match the name of the repo.
```

File: modules/ROOT/pages/design-storage-github.adoc [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+. Add the extracted content to the repository.
+. Import the project from GitHub.
+. Delete the original project.
+
+== Working with Branches
+
+Every time that you open a project that is saved in GitHub, API Designer retrieves the current content of the main branch. Every time that you switch to a different branch, API Designer retrieves the current content of that branch.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]: I suggest not switching between "main branch" and "default branch."
Try to stick with only one term for a thing.

Comment (kbostrom) [reply modules/ROOT/pages/design-storage-github.adoc]: Also, in both sentences, you could replace "Every time that you..." with "When you...".

File: modules/ROOT/pages/design-storage-github.adoc [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+
+The branch that you are currently working in is displayed at the top of the central pane of the text editor. Click the name of the branch to open a menu to carry out any of the following tasks:
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]: Try to replace even common idioms:
```suggestion
The branch that you are currently working in is displayed at the top of the central pane of the text editor. Click the name of the branch to open a menu to perform any of the following tasks:
```

File: modules/ROOT/pages/design-storage-github.adoc [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+
+* Switch to a different branch by clicking the name of that branch.
++
+Every time you open a project that is stored in GitHub as a repository, the content of the default branch appears in the text editor. So, if you want to work in a non-default branch, you must switch to that branch.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]: Do we need to repeat?

Comment (kbostrom) [reply modules/ROOT/pages/design-storage-github.adoc]:
```suggestion
Every time you open a project that is stored in GitHub as a repository, the content of the default branch appears in the text editor. So, if you want to work in a nondefault branch, you must switch to that branch.
```

File: modules/ROOT/pages/design-storage-github.adoc [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
++
+Before switching, you can view the difference between that branch and the current one.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]:
```suggestion
Before switching, you can view the difference between the branch you want to switch to and the current one.
```

File: modules/ROOT/pages/design-storage-github.adoc [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+After you create a branch, the text editor switches to that branch.
+* Open a pull request.
++
+This option behaves exactly as the *Create Pull Request* option in GitHub Desktop. It opens a page in GitHub where you can specify the name of your pull request and add a comment before clicking *Create*.
+* View the current branch on GitHub.
+* Delete a branch.
+
+== Managing Conflicting Changes
+
+If another user commits changes to the branch in which you are working, the text editor displays a dialog to inform you that is will discard your most recent changes (within the last 15 seconds or so) and incorporate the commits from the other user. Therefore, only one person at a time should work in a branch.
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]: ...and incorporate the changes
or
...and incorporate the commit

File: modules/ROOT/pages/design-storage-github.adoc [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+
+== Removing from Design Center an API Designer Project that is Saved in GitHub
```
Comment (kbostrom) [inline modules/ROOT/pages/design-storage-github.adoc]: Please use the Chicago tab of the online capitalization tool:
```suggestion
== Removing from Design Center an API Designer Project That Is Saved in GitHub
```

**Review by kbostrom — DISMISSED** [review]
Done

**Comment by failup** [comment]
We are closing this contribution because it has exceeded the age limit for a timely merge. If the contribution is still valid, please open a separate GitHub issue or pull request.

**Pull request closed without merging**